examples/api/lib/material/selection_area/custom_container.dart

```diff
@@ -0,0 +1,136 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// This sample demonstrates how to create a [SelectionContainer] that only
+// allows selecting everything or nothing with no partial selection.
+
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+
+void main() => runApp(const MyApp());
+
+class MyApp extends StatelessWidget {
+  const MyApp({super.key});
+
+  static const String _title = 'Flutter Code Sample';
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      title: _title,
+      home: SelectionArea(
```

Comment:
I don't think I remember this from my last read-through and before I forget: Do we document somewhere where one should introduce a SelectionArea if one wants to activate global selection? E.g. would it have been ok to just wrap the entire MaterialApp in one? Does this have to be done per route? Or something else?

Reply:
There are some documentations in both SelectableRegion and SelectionArea to talk about default behavior and how to use them